letsencrypt-auto-source/letsencrypt-auto.template

```diff
@@ -330,13 +330,20 @@ elif [ -f /etc/redhat-release ]; then
   prev_le_python="$LE_PYTHON"
   unset LE_PYTHON
   DeterminePythonVersion "NOCRASH"
+
   # Starting to Fedora 29, python2 is on a deprecation path. Let's move to python3 then.
+  # RHEL 8 also uses python3 by default.
   RPM_DIST_NAME=`(. /etc/os-release 2> /dev/null && echo $ID) || echo "unknown"`
-  RPM_DIST_VERSION=0
-  if [ "$RPM_DIST_NAME" = "fedora" ]; then
-    RPM_DIST_VERSION=`(. /etc/os-release 2> /dev/null && echo $VERSION_ID) || echo "0"`
-  fi
+  RPM_DIST_VERSION=`(. /etc/os-release 2> /dev/null && echo $VERSION_ID) | cut -d '.' -f1 || echo "0"`
```

Comment:
Testing this script on CentOS 6, I get the output:
```
./letsencrypt-auto: line 763: [: : integer expression expected
./letsencrypt-auto: line 765: [: : integer expression expected
Creating virtual environment...
You have an ancient version of Python entombed in your operating system...
This isn't going to work; you'll need at least version 2.7.
```
I think the problem here is in the case of a failure parsing `/etc/os-release` like on CentOS 6 where it doesn't exist, `RPM_DIST_VERSION` is being set to a string.

Reply:
Hell, it means that my "fail-safe" `|| echo "0"` instruction was not fail-safe at all, and would set `RPM_DIST_VERSION` to a string. It was working because this line was only executed on fedora, and fedora has an existing and correct `/etc/os-release`.

I think the fail-safe instruction should be `|| echo 0`.

Reply:
I've changed the failsafe to echo 0.  I can test on CentOS 6 when I return to work tomorrow.

Reply:
Just ran the script on a CentOS 6 VM, everything appears to be working.

Reply:
I see the problem.  The failsafe is never being evaluated because of the pipe to cut which returns as true even when the file being read doesn't exist.

```
[root@c6 ~]# cat /tmp/foo.txt 2>/dev/null | cut -d "." -f1
[root@c6 ~]# echo $?
0
```

I added a condition to check if /etc/os-release exists which now allows the script to run on CentOS 6.

Reply:
Another thing I just thought of, these conditions apply to more than just RHEL and CentOS.  There's also ScientificLinux and OracleLinux which are basically identical to RHEL.  Would it be acceptable to add a regex on line 347 to cover these other distros?  For example:

```
elif [ "$RPM_DIST_NAME" =~ "rhel|scientific|oracle|centos" -a "$RPM_DIST_VERSION" -ge 8 ]; then
```

Reply:
Technically it would. However, we saw that modifications done on `certbot-auto` were responsible of almost all point releases we had to do for the last 6 months.

Also the fact that `certbot-auto` should be removed in the next months in favor of a more robust and universal packaging approach is to be taken into accounts.

For these reasons, we want to avoid to do any modification to the script if it is not necessary to prevent a regression.

It is this case for that PR, since RHEL 7 is supported by `certbot-auto`, so having it failing when someone upgrade to RHEL 8 is definitely a regression.

However, adding new distributions, even really similar to the supported ones by `certbot-auto`, is really subject to discussions.

I think it should be not included in that PR for the least. But we can discuss about doing it or not for a future PR.

Reply:
Is there a list of what distros are officially supported?  According to the  system requirements page certbot only requires "Python 2.7 or 3.4+ running on a UNIX-like operating system."

I'll hold off on that change but it's definitely something to be considered for future releases.

Reply:
> Is there a list of what distros are officially supported?

Not really. There's a list of systems we offer explicit instructions in the dropdown menus at https://certbot.eff.org/instructions, however, in the same vein as your quoted text, there's an option there for "Other UNIX" which recommends that you use certbot-auto.

> I'll hold off on that change but it's definitely something to be considered for future releases.

I agree.